### ISSUE TITLE
Time filter for GPX fails if <metadata> has no <time>

### ISSUE THREAD

**Issue by rasher** [NathanBaulch/rainbow-roads#6]
It seems that `-after` and `-before` filtering (as well as the "period" console output) for GPX files relies on the presence of a <time> element in the <metadata> section, which isn't reliably there for all GPX files.

https://github.com/NathanBaulch/rainbow-roads/blob/bc6709fc90f5d97dc59c3557355aa5b7c01168d2/gpx.go#L77

I'd suggest the filtering and reading shouldn't actually be using the gpx metadata at all, but rather the earliest/latest recorded timestamps. Or at least do that as a backup.

**Comment by NathanBaulch** [comment]
Good catch, fixed. I can do a new release if you're unfamiliar with building Go from source, let me know.
